File: feature-history/src/main/java/com/andyanika/translator/feature/history/HistoryFragment.java
package com.andyanika.translator.feature.history;

import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import com.jakewharton.rxbinding4.InitialValueObservable;
import com.jakewharton.rxbinding4.widget.RxTextView;

import javax.inject.Inject;

import dagger.android.support.DaggerFragment;

public class HistoryFragment extends DaggerFragment {
    private final static int LIMIT = 100;

    @Inject
    HistoryListAdapter adapter;

    @Inject
    ViewModelProvider.Factory viewModelFactory;

    private HistoryViewModel viewModel;
    private EditText editInput;
    private View clearBtn;
    private InitialValueObservable<CharSequence> textObservable;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_history, null);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        editInput = view.findViewById(R.id.edit_input);
        textObservable = RxTextView.textChanges(editInput);
        clearBtn = view.findViewById(R.id.btn_clear);

        viewModel = ViewModelProviders.of(this, viewModelFactory).get(HistoryViewModel.class);
        viewModel.data.observe(this, adapter::setData);
        viewModel.showClearBtn.observe(this, show -> {
            if (show) {
                clearBtn.setVisibility(View.VISIBLE);
            } else {
                clearBtn.setVisibility(View.INVISIBLE);
                editInput.setText("");
            }
        });

        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
        RecyclerView recyclerView = view.findViewById(R.id.recycler_view);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void onStart() {
        super.onStart();
        clearBtn.setOnClickListener(v -> viewModel.showClearBtn.setValue(false));
        viewModel.subscribeSearch(textObservable, LIMIT);
        viewModel.subscribeItemClick(adapter.getObservable());
    }

    @Override
    public void onStop() {
        clearBtn.setOnClickListener(null);
        viewModel.unsubscribe();
        super.onStop();
    }
}

File: feature-history/src/main/kotlin/com/andyanika/translator/feature/history/HistoryFragment.kt
package com.andyanika.translator.feature.history

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.andyanika.translator.feature.history.databinding.FragmentHistoryBinding
import com.andyanika.usecases.di.koin.UseCaseComponent
import com.andyanika.widgets.observe
import core.models.FavoriteModel
import kotlinx.coroutines.flow.Flow
import org.koin.android.ext.android.inject
import org.koin.androidx.scope.ScopeFragment
import org.koin.androidx.viewmodel.ext.android.viewModel
import org.koin.core.context.loadKoinModules
import org.koin.core.parameter.parametersOf

class HistoryFragment : ScopeFragment() {
    init {
        loadKoinModules(UseCaseComponent.getModule())
    }

    private val vm by viewModel<HistoryViewModel>()
    private val action = { model: FavoriteModel -> vm.subscribeItemClick(model) }
    private val adapter by inject<HistoryListAdapter> { parametersOf(action) }

    private lateinit var binding: FragmentHistoryBinding

    private val inputTextFlow: Flow<String> by lazy {
        binding.editInput.observe()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_history, null)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding = FragmentHistoryBinding.bind(view)
        with(binding) {
            vm.data.observe(viewLifecycleOwner, adapter::setData)
            vm.showClearBtn.observe(viewLifecycleOwner, { show: Boolean ->
                if (show) {
                    btnClear.visibility = View.VISIBLE
                } else {
                    btnClear.visibility = View.INVISIBLE
                    editInput.setText("")
                }
            })

            recyclerView.layoutManager = LinearLayoutManager(context)
            recyclerView.adapter = adapter
        }
    }

    override fun onStart() {
        super.onStart()
        binding.btnClear.setOnClickListener { vm.showClearBtn.setValue(false) }
        vm.subscribeSearch(inputTextFlow, LIMIT)
    }

    override fun onStop() {
        binding.btnClear.setOnClickListener(null)
        super.onStop()
    }

    companion object {
        private const val LIMIT = 100
    }
}
